packages/medusa/src/interfaces/abstract-parser.ts

```diff
@@ -0,0 +1,30 @@
+import { ParseConfig } from "papaparse"
+
+export interface IParser<TResult> {
+  parse(path: string): Promise<TResult[]>
+}
+
+export type LineContext = {
+  lineNumber: number
+}
+
+export type ParserOptions = ParseConfig
+
+export abstract class AbstractParser<TSchema, TParserResult, TOutputResult>
+  implements IParser<TParserResult>
+{
+  protected $$schema: TSchema
+
+  protected constructor(schema: TSchema) {
+    this.$$schema = schema
+  }
+
+  public abstract parse(
+    path: string,
+    options?: ParserOptions
```

Comment:
issue: does the compiler works here? The options is not part of the parse method to implement from the interface

Reply:
haha for some reason it actually does, but well spotted, will fix 👍

Reply:
This is weird ^^ it not the first PR where the build pass but it shouldn’t